rust/src/metadata.rs

```diff
@@ -247,16 +248,14 @@ impl TransactionMetadatumLabels {
 
 #[wasm_bindgen]
 #[derive(Clone, Debug, Eq, Ord, PartialEq, PartialOrd)]
-pub struct TransactionMetadata(
-    linked_hash_map::LinkedHashMap<TransactionMetadatumLabel, TransactionMetadatum>,
-);
+pub struct GeneralTransactionMetadata(LinkedHashMap<TransactionMetadatumLabel, TransactionMetadatum>);
 
-to_from_bytes!(TransactionMetadata);
+to_from_bytes!(GeneralTransactionMetadata);
 
 #[wasm_bindgen]
-impl TransactionMetadata {
+impl GeneralTransactionMetadata {
```

Comment:
I thought about having this stay in the same struct as before and simply adding a `native_scripts()` and `set_native_scripts()` field to it to make it so the API stays the same for anyone not using the new native scripts metadata but then the `insert()` , `len()` and `get()` might be confusing whether they refer to the old metadata instead of indexing/getting the length of the native script portion. We could do that anyway if we want to not change the API. I didn't use the direct code-gen since that would've been even more structs generated (eg `TransactionMetadataEnum`, `TransactionMetadataKind`, and `AllegraMetadata` with more indirection and it being treated like the other enums) but since one is basically a subset of the other I combined them somewhat.

Reply:
Yeah, I think this makes sense. I'm slightly concerned that the fact we're keeping the `TransactionMetadata` name (but not referring to something different) may cause subtle bugs for people though.